type-c-service/src/wrapper/backing.rs

```diff
@@ -102,20 +102,30 @@ struct InternalState<'a, const N: usize> {
 }
 
 impl<'a, const N: usize> InternalState<'a, N> {
-    fn new<M: RawMutex>(storage: &'a Storage<N, M>) -> Self {
-        Self {
-            controller_state: ControllerState::default(),
-            port_states: from_fn(|i| PortState {
+    fn new<M: RawMutex>(storage: &'a Storage<N, M>) -> Option<Self> {
+        let port_states = from_fn(|i| {
+            Some(PortState {
                 status: PortStatus::new(),
                 sw_status_event: PortStatusChanged::none(),
                 sink_ready_deadline: None,
                 pending_events: PortEvent::none(),
                 pd_alerts: (
-                    storage.pd_alerts[i].dyn_immediate_publisher(),
-                    storage.pd_alerts[i].dyn_subscriber().unwrap(),
+                    storage.pd_alerts.get(i)?.dyn_immediate_publisher(),
+                    storage.pd_alerts.get(i)?.dyn_subscriber().ok()?,
                 ),
-            }),
+            })
+        });
```

Comment:
This is `array::map`
```suggestion
        let port_states = storage.pd_alerts.map(|pd_alert| {
            Some(PortState {
                status: PortStatus::new(),
                sw_status_event: PortStatusChanged::none(),
                sink_ready_deadline: None,
                pending_events: PortEvent::none(),
                pd_alerts: (
                    pd_alerts.dyn_immediate_publisher(),
                    pd_alerts.dyn_subscriber().ok()?,
                ),
            })
        });
```